gef.py

```diff
@@ -136,80 +125,92 @@ def update_gef(argv):
-__commands__                           = []
-__functions__                          = []
-__aliases__                            = []
-__config__                             = {}
-__watches__                            = {}
-__infos_files__                        = []
-__gef_convenience_vars_index__         = 0
-__context_messages__                   = []
-__heap_allocated_list__                = []
-__heap_freed_list__                    = []
-__heap_uaf_watchpoints__               = []
-__pie_breakpoints__                    = {}
-__pie_counter__                        = 1
-__gef_remote__                         = None
-__gef_qemu_mode__                      = False
-__gef_current_arena__                  = "main_arena"
-__gef_int_stream_buffer__              = None
-__gef_redirect_output_fd__             = None
+GDB_MIN_VERSION                        = (8, 0)
+GDB_VERSION                            = tuple(map(int, re.search(r"(\d+)[^\d]+(\d+)", gdb.VERSION).groups()))
+PYTHON_MIN_VERSION                     = (3, 6)
+PYTHON_VERSION                         = sys.version_info[0:2]
 
 DEFAULT_PAGE_ALIGN_SHIFT               = 12
 DEFAULT_PAGE_SIZE                      = 1 << DEFAULT_PAGE_ALIGN_SHIFT
-GEF_RC                                 = os.getenv("GEF_RC") or os.path.join(os.getenv("HOME"), ".gef.rc")
+
+GEF_RC                                 = pathlib.Path(os.getenv("GEF_RC")).absolute() if os.getenv("GEF_RC") else pathlib.Path().home() / ".gef.rc"
 GEF_TEMP_DIR                           = os.path.join(tempfile.gettempdir(), "gef")
 GEF_MAX_STRING_LENGTH                  = 50
 
-GDB_MIN_VERSION                        = (7, 7)
-GDB_VERSION                            = tuple(map(int, re.search(r"(\d+)[^\d]+(\d+)", gdb.VERSION).groups()))
+LIBC_HEAP_MAIN_ARENA_DEFAULT_NAME      = "main_arena"
+ANSI_SPLIT_RE                          = r"(\033\[[\d;]*m)"
 
-current_elf  = None
-current_arch = None
+LEFT_ARROW                             = " \u2190 "
+RIGHT_ARROW                            = " \u2192 "
+DOWN_ARROW                             = "\u21b3"
+HORIZONTAL_LINE                        = "\u2500"
+VERTICAL_LINE                          = "\u2502"
+CROSS                                  = "\u2718 "
+TICK                                   = "\u2713 "
+BP_GLYPH                               = "\u25cf"
+GEF_PROMPT                             = "gef\u27a4  "
+GEF_PROMPT_ON                          = f"\001\033[1;32m\002{GEF_PROMPT}\001\033[0m\002"
+GEF_PROMPT_OFF                         = f"\001\033[1;31m\002{GEF_PROMPT}\001\033[0m\002"
 
-libc_args_definitions = {}
 
-highlight_table = {}
-ANSI_SPLIT_RE = r"(\033\[[\d;]*m)"
+gef : "Gef"                                                           = None
+__registered_commands__ : List["GenericCommand"]                      = []
+__registered_functions__ : List["GenericFunction"]                    = []
+__registered_architectures__ : Dict[Union[int, str], "Architecture"]  = {}
```

Comment:
I think the type hint is wrong. It should be `Dict[Union[int, str], Type]`
because each architecture is a class. setting it to Architecture implies is an INSTANCE of Architecture, not a subclass object